src/NuGet.CommandLine/Commands/DownloadCommandBase.cs

```diff
@@ -105,9 +105,22 @@ protected IEnumerable<Configuration.PackageSource> GetPackageSources(Configurati
             var availableSources = SourceProvider.LoadPackageSources().Where(source => source.IsEnabled);
             var packageSources = new List<Configuration.PackageSource>();
 
-            if (!NoCache && !string.IsNullOrEmpty(MachineCache.Default?.Source))
+            if (!NoCache)
             {
-                packageSources.Add(new V2PackageSource(MachineCache.Default.Source, () => MachineCache.Default));
+                // Add the v2 machine cache
+                if (!string.IsNullOrEmpty(MachineCache.Default?.Source))
+                {
+                    packageSources.Add(new V2PackageSource(MachineCache.Default.Source, () => MachineCache.Default));
+                }
+
+                // Add the v3 global packages folder
+                var globalPackageFolder = SettingsUtility.GetGlobalPackagesFolder(settings);
+
+                if (!string.IsNullOrEmpty(globalPackageFolder) && Directory.Exists(globalPackageFolder))
+                {
+                    packageSources.Add(new V2PackageSource(globalPackageFolder, 
+                        () => new LocalPackageRepository(globalPackageFolder)));
```

Comment:
I believe that globalPackagesFolder cannot be used as a LocalPackageRepository, because, it only tries to look for packages 1 level deep. Meaning at the <repo> folder and at the immediate subdirectories of <repo> folder.